opentracing/src/main/java/opentracing/TraceCodec.java

```diff
@@ -24,16 +24,16 @@
  * binary message format, so that it can be inspected efficiently by the middleware / routing layers
  * without parsing the whole message.
  */
-public interface TraceContextCodec {
+public interface TraceCodec {
```

Comment:
@michaelsembwever @adriancole 

Most organizations I've spoken with do most of their IPC via ad hoc HTTP calls. By including a codec facility in the API, those ad hoc HTTP calls propagate the TraceContext (or SpanContext, here) from process to process *without relying on knowledge of the particular OpenTracing implementation*. Breaking that is a non-starter.